Use c10::variant-based enums for SmoothL1Loss module and functional

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29673 Use F::*FuncOptions for embedding/embeddingbag functionals
* #29584 Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h
* **#29536 Use c10::variant-based enums for SmoothL1Loss module and functional**
* #29535 Use c10::variant-based enums for F::grid_sample
* #29404 Make all non-input arguments to functionals part of its options, and pass functional options by const-ref when possible

Differential Revision: [D18432272](https://our.internmc.facebook.com/intern/diff/D18432272)

Commit:
Use c10::variant-based enums for SmoothL1Loss module and functional

Commit:
Update on "Use c10::variant-based enums for SmoothL1Loss module and functional"

Use c10::variant-based enums for SmoothL1Loss module and functional

gh-metadata: pytorch pytorch 29536 gh/yf225/44/head

kostmo commented:
## CircleCI build failures summary
As of commit bebb692:
* **1/1** failures introduced in this PR
* **0/1** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=bebb69204a4521e0e3b031fef411613aaae56f31).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

Commit:
Update on "Use c10::variant-based enums for SmoothL1Loss module and functional"

Use c10::variant-based enums for SmoothL1Loss module and functional

gh-metadata: pytorch pytorch 29536 gh/yf225/44/head

Commit:
Update on "Use c10::variant-based enums for SmoothL1Loss module and functional"

Use c10::variant-based enums for SmoothL1Loss module and functional

gh-metadata: pytorch pytorch 29536 gh/yf225/44/head

Commit:
Update on "Use c10::variant-based enums for SmoothL1Loss module and functional"

Use c10::variant-based enums for SmoothL1Loss module and functional

gh-metadata: pytorch pytorch 29536 gh/yf225/44/head
